DiffusionPreprocessing/scripts/eddy_postproc.sh

```diff
@@ -87,13 +87,31 @@ else
 fi
 #fi
 
+
+#Create a mask representing voxels within the field of view for all volumes prior to dilation
+${FSLDIR}/bin/fslmaths ${datadir}/data -abs -Tmin -bin -fillh ${datadir}/fov_mask
+
 	 
 if [ ! $GdCoeffs = "NONE" ] ; then
     echo "Correcting for gradient nonlinearities"
 	# Note: "data_warped" is eddy-current and suspectibility distortion corrected (via 'eddy'), but prior to gradient distortion correction
 	# i.e., "data_posteddy_preGDC" would be another way to think of it
     ${FSLDIR}/bin/immv ${datadir}/data ${datadir}/data_warped
-    ${globalscriptsdir}/GradientDistortionUnwarp.sh --workingdir="${datadir}" --coeffs="${GdCoeffs}" --in="${datadir}/data_warped" --out="${datadir}/data" --owarp="${datadir}/fullWarp"
+
+    # Dilation outside of the field of view to minimise the effect of the hard field of view edge on the interpolation
+	  DiffRes=`${FSLDIR}/bin/fslval ${datadir}/data pixdim1`
+    DilateDistance=`echo "$DiffRes * 4" | bc`  # Extrapolates the diffusion data up to 4 voxels outside of the FOV
+    ${CARET7DIR}/wb_command -volume-dilate ${datadir}/warped/data_warped.nii.gz $DilateDistance NEAREST ${datadir}/warped/data_warped_dilated.nii.gz
+
+    # apply gradient distortion correction
+    ${globalscriptsdir}/GradientDistortionUnwarp.sh --workingdir="${datadir}" --coeffs="${GdCoeffs}" --in="${datadir}/data_warped_dilated" --out="${datadir}/data" --owarp="${datadir}/fullWarp"
+
+    # Transform field of view mask (using conservative trilinear interolation with high threshold)
+    ${FSLDIR}/bin/immv ${datadir}/fov_mask ${datadir}/fov_mask_warped
+    ${FSLDIR}/bin/applywarp --rel --interp=trilinear -i ${datadir}/fov_mask_warped -r ${datadir}/fov_mask_warped -w ${datadir}/fullWarp -o ${datadir}/fov_mask
+    ${FSLDIR}/bin/fslmaths ${datadir}/fov_mask -thr 0.999 -bin ${datadir}/fov_mask
+    # mask out any data outside the field of view
+    ${FSLDIR}/bin/fslmaths ${datadir}/data -mas ${datadir}/fov_mask ${datadir}/data
```

Comment:
I guess that technically it shouldn't be necessary to apply the fov_mask to the data in the `$GdCoeffs=NONE` condition, but for robustness of code I'd move the final masking operation to outside of the `$GdCoeffs` conditional.